internal/runner/operation.go

```diff
@@ -115,10 +115,15 @@ func (r *Runner) executeJob(
 	}
 
 	if path == "" {
-		// No waypoint.hcl file is found.
-		return nil, status.Errorf(codes.FailedPrecondition,
-			"A waypoint.hcl was not found. Please either add a waypoint.hcl to "+
-				"the project source or in the project settings in the Waypoint UI.")
+		// Only warn if operation is project destroy
+		if _, ok := job.Operation.(*pb.Job_DestroyProject); ok {
+			log.Warn("A waypoint.hcl was not found.")
```

Comment:
I don't know about this fix, we get around this error, but it only leads to more:

```
brian@ubuntu:~ λ waypoint project destroy -project=delete-me
Do you really want to destroy project "delete-me" and its resources? Only 'yes' will be accepted to approve: yes


» Performing operation locally
! Error destroying project: rpc error: code = Unknown desc = <nil>: Failed to read configuration; Can't read /home/brian: read /home/brian: is a directory.
```

Or if you run it in a different Waypoint project, it fails with no data source configured because the destroy operations can't run:

```
brian@ubuntu:waypoint-tetris λ waypoint project destroy -project=delete-me                                                                                ±[main]
Warning: Currently in project directory for "k8s-tetris", but will operate
against specified project "delete-me"
Do you really want to destroy project "delete-me" and its resources? Only 'yes' will be accepted to approve: yes

! Error destroying project: rpc error: code = FailedPrecondition desc = Project delete-me does not have a data source configured. Remote jobs require a data
source such as Git to be configured with the project. Data sources can be configured via the CLI or UI. For help, see :
https://www.waypointproject.io/docs/projects/git#configuring-the-project
brian@ubuntu:waypoint-tetris λ                                            
```

I think instead we should have a flow to support deleting projects with no data source supported via the Destroy operation in the case where we have a blank project.